packages/types/src/address_provider.rs

```diff
@@ -146,28 +146,43 @@ pub mod helpers {
 
     use super::{AddressResponseItem, MarsAddressType, QueryMsg};
 
-    pub fn query_address(
+    /// Query contract address.
+    ///
+    /// It fails if the provided address does not start with current chain prefix.
+    pub fn query_contract_addr(
         deps: Deps<impl cosmwasm_std::CustomQuery>,
         address_provider_addr: &Addr,
         contract: MarsAddressType,
     ) -> StdResult<Addr> {
         let res: AddressResponseItem =
             deps.querier.query_wasm_smart(address_provider_addr, &QueryMsg::Address(contract))?;
-
         deps.api.addr_validate(&res.address)
```

Comment:
Imo this syntax is cleaner

```suggestion
        deps.querier
            .query_wasm_smart::<AddressResponseItem>(
                address_provider_addr,
                &QueryMsg::Address(contract),
            )
            .map(|res| deps.api.addr_validate(&res.address))?
```

Reply:
ok ;)

Reply:
```Applying suggestions on deleted lines is currently not supported. Please  and try again.```

Reply:
@piobab I pushed a commit